syntax = "proto3";

package ray.rpc;

// The greeting service definition.
service Failover {
  // Sends a greeting
  rpc ResetState (ResetStateRequest) returns (ResetStateReply) {}
}

// The request message containing the user's name.
message ResetStateRequest {
  uint64 secret = 1;
}

// The response message containing the greetings
message ResetStateReply {
}

// This enum is needed by asio based rpc. For each method in the rpc service,
// a message type needs to be defined for the corresponding request in that
// method, so that receiver side can figure out the exact request message
// based on this type, deserialize it accordingly, and invoke the corresponding
// method.
enum FailoverMessageType {
  // `ConnectClient` and `DisconnectClient` MUST be defined and their values
  // MUST be consistent with the ones in `ServiceMessageType` in
  // "src/ray/protobuf/asio.proto".
  FailoverConnectClient = 0;
  FailoverDisconnectClient = 1;
  // Push task request.
  ResetStateRequestMessage = 2;
  // Push task reply.
  ResetStateReplyMessage = 3;
}